docs/developer/release-process.md

```diff
@@ -2,67 +2,154 @@
+
+> As part of this release process, the dashboard deps _must_ be updated, the golang deps _should_ be updated, the rust deps _should_ be updated and the security check _must_ be performed.
+
+## 1 - Send a PR to the bitnami/chart repository
+
+Since the chart that we host in the Kubeapps repository is only intended for development purposes, we need to synchronize it with the official one in the [bitnami/charts repository](https://github.com/bitnami/charts/tree/master/bitnami/kubeapps). To this end, we need to send a PR with the changes to their repository and wait until it gets accepted.
+
+> This step is currently manual albeit prone to change shortly.
+
+## 2 - Create a new git tag
+
+Once the dependencies have been updated and the chart changes merged, the next step is to tagging the latest commit in master and pushing it to the main branch. Please note that the tag name will be used as the release name.
+
+For doing so, execute the following commands:
+
+```bash
+export VERSION_NAME="v1.0.0-beta.1" # edit it accordingly
 
 git tag ${VERSION_NAME}
 git push origin ${VERSION_NAME}
 ```
 
-This will trigger a build, test and **release** [workflow in our CI](https://circleci.com/gh/kubeapps/workflows).
-
-## 2 - Complete the GitHub release notes
+A new tag pushed to the repository will trigger, apart from the usual test and build steps, a _release_ [workflow](https://circleci.com/gh/kubeapps/workflows) as described in the [CI documentation](./ci.md).
 
-Once the release job is finished, you will have a GitHub release draft pre-populated. You still must **add a high level description with the release highlights**. Save the draft and **do not publish it yet**.
+> When a new tag is detected, Bitnami will automatically build a set of container images based on the tagged commit. They later will be published in [the Dockerhub image registry](https://hub.docker.com/search?q=bitnami%2Fkubeapps&type=image).
 
-## 3 - Bump the chart version
+## 3 - Complete the GitHub release notes
 
-At this point, you will have a new set of published docker images as well as some release notes waiting to be published.
+Once the release job is finished, you will have a pre-populated [draft GitHub release](https://github.com/kubeapps/kubeapps/releases).
 
-But before, we need to create and merge a PR with a chart version bump in `chart/kubeapps/Chart.yaml` ([example](https://github.com/kubeapps/kubeapps/pull/663/files)). This will trigger another CI job that will publish a new version of the chart pointing to the new Docker images built in the step 1.
+You still must **add a high-level description with the release highlights**. Save the draft and **do not publish it yet**. Then, get these notes reviewed by another Kubeapps maintainer.
```

Comment:
Might be worth mentioning to at least separate out all the version update commits?